consensus/src/chained_bft/event_processor.rs

```diff
@@ -740,8 +745,19 @@ impl<T: Payload> EventProcessor<T> {
     /// 1) fetch missing dependencies if required, and then
     /// 2) call process_certificates(), which will start a new round in return.
     async fn add_vote(&mut self, vote: &Vote) -> anyhow::Result<()> {
+        let block_id = vote.vote_data().proposed().id();
         // Add the vote and check whether it completes a new QC or a TC
-        match self.block_store.insert_vote(vote, &self.validators) {
+        let res = self.pending_votes.insert_vote(vote, &self.validators);
```

Comment:
The way it's implemented here we're going to trigger the `new_qc_aggregated()` flow even after the QC has been created. We should check that the QC is not already present in the block store prior to adding it to the pending votes.

Reply:
It's also fine to reject in PendingVotes if it already has enough votes

Reply:
Addressed at 750 - 752